src/ParseQuery.js

```diff
@@ -1005,21 +1005,49 @@ class ParseQuery {
   *
   * @param {String} key The key that the string to match is stored in.
   * @param {String} value The string to search
+  * @param {Object} options (Optional) See https://docs.mongodb.com/manual/reference/operator/query/text/#text-query-operator-behavior
+  * @param {String} options.language
+  * @param {String} options.caseSensitive
+  * @param {String} options.diacriticSensitive
   * @return {Parse.Query} Returns the query, so you can chain this call.
   */
- fullText(key: string, value: string): ParseQuery {
-   if (!key) {
-     throw new Error('A key is required.');
-   }
-   if (!value) {
-     throw new Error('A search term is required');
-   }
-   if (typeof value !== 'string') {
-     throw new Error('The value being searched for must be a string.');
-   }
-
-   return this._addCondition(key, '$text', { $search: { $term: value } });
- }
+  fullText(key: string, value: string, options: ?Object): ParseQuery {
+    options = options || {};
+
+    var fullOptions = {};
+    for (var key in options) {
+      switch (key) {
+        case 'language':
+          fullOptions.$language = options[key];
+          break;
+        case 'caseSensitive':
+          fullOptions.$caseSensitive = options[key];
+          break;
+        case 'diacriticSensitive':
+          fullOptions.$diacriticSensitive = options[key];
+          break;
+        default:
+          throw new Error('Unknown option: '+key);
+          break;
+      }
+    }
+
+    if (options && options.hasOwnProperty('$diacriticSensitive')) {
```

Comment:
Remove this as it makes all your test fail. Its basically saying `$diacriticSensitive` is required

Were you able to run test locally.

Reply:
oops =|